### ISSUE TITLE
Feasibility of adapting Protractor to other frameworks

### ISSUE THREAD

**Issue by bclinkinbeard** [angular/protractor#1507]
**TL;DR** Could changing `waitForAngular` enable Protractor to test other frameworks like Ember?

Custom locators dealing with models and bindings are obviously specific to Angular, but IMO Protractor's API is far superior to any of the other selenium-webdriver wrappers. I know about `browser.ignoreSynchronization = true;`, but that sort of short-circuits a lot of Protractor's ability to wait intelligently.

As far as I can tell, Protractor simply uses `angular.element(el).injector().get('$browser').notifyWhenNoOutstandingRequests(callback);` from https://github.com/angular/protractor/blob/master/lib/clientsidescripts.js#L34 to know when it's OK to move on to the next step. Essentially, my question is whether Protractor would work if I were able to change that function to something analogous for another framework like Ember, or are there other things specific to Angular that would result in additional roadblocks.
